server/room.ts

```diff
@@ -973,15 +986,28 @@ export class Room {
         return;
       }
     }
+    if (roomTitle && roomTitle.length > 50) {
+      return;
+    }
+    if (roomDescription && roomDescription.length > 120) {
+      return;
+    }
+    // check if is valid hex color representation
+    if (!/^#([0-9a-f]{3}){1,2}$/i.test(roomTitleColor)) {
+      return;
+    }
     // console.log(owner, vanity, password);
     const roomObj: any = {
       roomId: this.roomId,
       password: password,
       isChatDisabled: isChatDisabled,
+      roomTitle: roomTitle,
+      roomDescription: roomDescription,
     };
     if (isSubscriber) {
-      // user must be sub to set vanity
+      // user must be sub to set certain properties
```

Comment:
should we make this whole feature sub-only?

Reply:
OK, this was my original intention also.